projects/schematics/src/shared/utils/package-utils.ts

```diff
@@ -51,3 +53,25 @@ export function isAngularLocalizeInstalled(tree: Tree): boolean {
   const packageJsonObject = readPackageJson(tree);
   return packageJsonObject.dependencies.hasOwnProperty(ANGULAR_LOCALIZE);
 }
+
+export function checkIfSSRIsUsed(tree: Tree): boolean {
+  const projectName = getDefaultProjectNameFromWorkspace(tree);
+  const buffer = tree.read('angular.json');
+  if (!buffer) {
+    throw new SchematicsException('Could not find angular.json');
+  }
+  const angularFileBuffer = buffer.toString(UTF_8);
+  const angularJson = JSON.parse(angularFileBuffer);
+  const isServerConfiguration = !!angularJson.projects[projectName].architect[
+    'server'
+  ];
+  const serverFilePath = getPathResultsForFile(tree, 'server.ts', '/')[0];
```

Comment:
Apparently some customers have their server in a file not named `server.ts` or in a different location. For example, it could be `server/index.ts`.

This file can be found in the `angular.json` "server configuration" in "options" -> "main". Maybe we could use that value instead of a hard-coded file path.

What do you think?

Reply:
Good point, let's do that.